src/openzaak/components/documenten/tests/test_upload.py

```diff
@@ -0,0 +1,1007 @@
+        )
+
+        response = self.client.patch(
+            eio_url, {"beschrijving": "beschrijving2", "lock": self.canonical.lock}
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        self.canonical.refresh_from_db()
+        part_new = self.canonical.bestandsdelen.order_by("volgnummer").first()
+
+        self.assertEqual(self.canonical.bestandsdelen.count(), 2)
+        self.assertEqual(self.canonical.empty_bestandsdelen, True)
+        self.assertEqual(part_new.voltooid, False)
+
+    def test_update_metadata_set_size(self):
+        """
+        Test the update process of the file size with some of part files uploaded
+        Input:
+        * bestandsomvang > 0
+
+        Expected result:
+        * bestandsdelen objects are regenerated based on the new bestandsomvang
+        * all uploaded part files are lost
+        """
+        self._create_metadata()
+
+        # update file metadata
+        eio_url = get_operation_url(
+            "enkelvoudiginformatieobject_read", uuid=self.eio.uuid
+        )
+
+        response = self.client.patch(
+            eio_url, {"bestandsomvang": 45, "lock": self.canonical.lock}
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        data = response.json()
+        self.canonical.refresh_from_db()
+        new_version = self.canonical.latest_version
+
+        self.assertEqual(new_version.bestandsomvang, 45)
+        self.assertEqual(self.canonical.bestandsdelen.count(), 5)
+        self.assertEqual(data["inhoud"], None)
+
+    def test_update_metadata_set_size_zero(self):
+        """
+        Test the update process of the file size = 0
+        Input:
+        * bestandsomvang = 0
+
+        Expected result:
+        * bestandsdelen objects are removed
+        * empty file is created
+        * file link points to this empty file
+        """
+        self._create_metadata()
+
+        # update file metadata
+        eio_url = get_operation_url(
+            "enkelvoudiginformatieobject_read", uuid=self.eio.uuid
+        )
+
+        response = self.client.patch(
+            eio_url, {"bestandsomvang": 0, "lock": self.canonical.lock}
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        data = response.json()
+        self.canonical.refresh_from_db()
+        new_version = self.canonical.latest_version
+        # file_url = get_operation_url(
+        #     "enkelvoudiginformatieobject_download", uuid=new_version.uuid
+        # )
+
+        self.assertEqual(new_version.bestandsomvang, 0)
+        self.assertEqual(self.canonical.bestandsdelen.count(), 0)
+        self.assertEqual(data["inhoud"], None)
+
+        # TODO: not sure why this was present in the original test, I would expect
+        # `inhoud` to be empty, because the file is empty as well
+        # self.assertEqual(data["inhoud"], f"http://testserver{file_url}?versie=1")
```

Comment:
I think this is the distinction between `0` and `None` - `0` means there is an empty file, so there is a file download URL.